plugins/inputs/hugepages/hugepages_test.go

```diff
@@ -0,0 +1,207 @@
+//go:build linux
+// +build linux
+
+package hugepages
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf/testutil"
+)
+
+func TestInit(t *testing.T) {
+	t.Run("when no config is provided then all fields should be set to default values", func(t *testing.T) {
```

Comment:
Usually we split this into multiple `Test...` functions. Is there a reason you do it this way?

Reply:
Yes, I think it is more organized to have one function per exported method and `t.Run`'s for every scenario (with self explanatory names). Are you strongly against?

Reply:
Personally, I better like the organization in groups of _what_ to test instead of which function to test, but I don't have strong feelings about this...